EmptyLineSeparator: false positive "',' should be separated from previous line" on multi-variable field declaration

I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyLineSeparator"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > Example.java <<'EOF'
class Example {
    int a = 1, b = 2;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml Example.java

Starting audit...
[ERROR] /Users/vivek/checkstyle/Example.java:2:14: ',' should be separated from previous line. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

A single multi-variable field declaration such as `int a = 1, b = 2;` should not produce an EmptyLineSeparator violation on the comma.
`a` and `b` are two `VARIABLE_DEF` nodes joined by a comma within one declaration. With the default `allowNoEmptyLineBetweenFields = false`, the check treats them as two fields that must be separated by an empty line and flags the comma (`isViolatingEmptyLineBetweenFieldsPolicy` special-cases the comma; setting `allowNoEmptyLineBetweenFields = true` suppresses it).  The requirement cannot be satisfied, you cannot put a blank line inside a single declaration.

Expected: a multi-variable declaration on one line should not be flagged on the comma by default.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java
@@ -487,8 +487,9 @@ public class EmptyLineSeparatorCheck extends AbstractCheck {
      */
     private boolean isViolatingEmptyLineBetweenFieldsPolicy(DetailAST detailAST) {
         return detailAST.getType() != TokenTypes.RCURLY
+                && detailAST.getType() != TokenTypes.COMMA
                 && (!allowNoEmptyLineBetweenFields
-                    || !TokenUtil.isOfType(detailAST, TokenTypes.COMMA, TokenTypes.VARIABLE_DEF));
+                    || detailAST.getType() != TokenTypes.VARIABLE_DEF);
     }
 
     /**

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheckTest.java
@@ -254,6 +254,23 @@ public class EmptyLineSeparatorCheckTest
     }
 
     @Test
+    public void testMultiVariableDeclaration() throws Exception {
+        final String[] expected = {
+            "22:5: " + getCheckMessage(MSG_SHOULD_BE_SEPARATED, "VARIABLE_DEF"),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputEmptyLineSeparatorMultiVariableDeclaration.java"), expected);
+    }
+
+    @Test
+    public void testMultiVariableDeclarationAllowNoEmptyLine() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWithInlineConfigParser(
+                getPath("InputEmptyLineSeparatorMultiVariableDeclarationAllowNoEmptyLine.java"),
+                expected);
+    }
+
+    @Test
     public void testAllowMultipleImportSeparatedFromPackage() throws Exception {
         final String[] expected = {
             "13:78: " + getCheckMessage(MSG_SHOULD_BE_SEPARATED, "import"),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorMultiVariableDeclaration.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorMultiVariableDeclaration.java
@@ -1,0 +1,23 @@
+/*
+EmptyLineSeparator
+allowNoEmptyLineBetweenFields = (default)false
+allowMultipleEmptyLines = (default)true
+allowMultipleEmptyLinesInsideClassMembers = (default)true
+tokens = (default)PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF, \
+         STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF, RECORD_DEF, \
+         COMPACT_CTOR_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator;
+
+public class InputEmptyLineSeparatorMultiVariableDeclaration {
+    int a = 1, b = 2;
+
+    int c = 3,
+        d = 4;
+
+    int e = 5;
+    int f = 6; // violation ''VARIABLE_DEF' should be separated from previous line.'
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorMultiVariableDeclarationAllowNoEmptyLine.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorMultiVariableDeclarationAllowNoEmptyLine.java
@@ -1,0 +1,23 @@
+/*
+EmptyLineSeparator
+allowNoEmptyLineBetweenFields = true
+allowMultipleEmptyLines = (default)true
+allowMultipleEmptyLinesInsideClassMembers = (default)true
+tokens = (default)PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF, \
+         STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF, RECORD_DEF, \
+         COMPACT_CTOR_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator;
+
+public class InputEmptyLineSeparatorMultiVariableDeclarationAllowNoEmptyLine {
+    int a = 1, b = 2;
+
+    int c = 3,
+        d = 4;
+
+    int e = 5;
+    int f = 6;
+}

